### PR TITLE
Add measured metric tag

### PR DESCRIPTION
**Pull request by delner**
To support better service statistics, this pull request adds the `_dd.measured` metric to spans that are "internal" to the host service.

### CODE REVIEW

**Comment by delner** [comment]
Just as a note, this implements it for all integrations except GraphQL which is managed in the GraphQL repo. We'll need to update this behavior in a separate PR for that repo.

File: docker-compose.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ services:
       - mongodb
       - mysql
       - postgres
+      - presto
```
Comment (marcotc) [inline docker-compose.yml]: https://github.com/DataDog/dd-trace-rb/pull/975 seems to address this, I think a rebase will like address the changes in this file.

File: lib/ddtrace/analytics.rb [hunk 2]
```diff
@@ -8,6 +8,12 @@ def set_sample_rate(span, sample_rate)
         return if span.nil? || !sample_rate.is_a?(Numeric)
         span.set_metric(Datadog::Ext::Analytics::TAG_SAMPLE_RATE, sample_rate)
       end
+
+      def set_measured(span, value = true)
+        return if span.nil?
+        value = [true, 1].include?(value) ? 1 : 0
```
Comment (marcotc) [inline lib/ddtrace/analytics.rb]: We likely want to extract `[true, 1]` into a constant, so that a new array is not allocated on each method call.

Comment (delner) [reply lib/ddtrace/analytics.rb]: Yeah, I thought exactly the same thing. Originally I had it in two `==` clauses instead, but Rubocop complained. I think I'm going to revert this back to that original clause and just disable the Rubocop rule, as the constant feels overkill.

File: lib/ddtrace/contrib/action_view/events/render_partial.rb [hunk 3]
```diff
@@ -31,6 +31,9 @@ def process(span, _event, _id, payload)
               span.set_tag(Ext::TAG_TEMPLATE_NAME, template_name)
             end
 
+            # Measure service stats
+            Contrib::Analytics.set_measured(span)
```
Comment (brettlangdon) [inline lib/ddtrace/contrib/action_view/events/render_partial.rb]: do we set a unique service name on action view today?

just curious about whether we generate metrics today for view rendering and if not, do we want to add metrics for view rendering from this?

File: lib/ddtrace/contrib/action_view/events/render_template.rb [hunk 4]
```diff
@@ -34,6 +34,9 @@ def process(span, _event, _id, payload)
             layout = payload[:layout]
             span.set_tag(Ext::TAG_LAYOUT, layout) if layout
 
+            # Measure service stats
+            Contrib::Analytics.set_measured(span)
```
Comment (brettlangdon) [inline lib/ddtrace/contrib/action_view/events/render_template.rb]: assuming the same question here as `render_partial.rb`

File: lib/ddtrace/contrib/action_view/instrumentation/partial_renderer.rb [hunk 5]
```diff
@@ -37,6 +37,9 @@ def datadog_render_partial(template)
                 Ext::TAG_TEMPLATE_NAME,
                 template_name
               )
+
+              # Measure service stats
+              Contrib::Analytics.set_measured(active_datadog_span)
```
Comment (brettlangdon) [inline lib/ddtrace/contrib/action_view/instrumentation/partial_renderer.rb]: would this get set by `render_partial.rb` `#process`?

File: lib/ddtrace/contrib/sinatra/tracer.rb [hunk 6]
```diff
@@ -42,6 +42,9 @@ def render(engine, data, *)
                   # want to record it.
                   span.set_tag(Ext::TAG_TEMPLATE_NAME, data) if data.is_a? Symbol
 
+                  # Measure service stats
+                  Contrib::Analytics.set_measured(span)
```
Comment (brettlangdon) [inline lib/ddtrace/contrib/sinatra/tracer.rb]: same here for action view, do we want to add metrics for template rendering?

Comment (brettlangdon) [reply lib/ddtrace/contrib/sinatra/tracer.rb]: or well, the question is partially "do we do it today?" + "does it make sense to add it going forward?"

File: lib/ddtrace/contrib/action_view/events/render_partial.rb (hunk 3 above)
Comment (delner) [reply lib/ddtrace/contrib/action_view/events/render_partial.rb]: I don't think we do, but I think we should. I think the basic components of request timing generally breaks down into:

1. Request queuing (provided by Rack instrumentation)
2. Middleware (nothing provided)
3. Processing time (provided by ActionPack instrumentation and others like Grape/GraphQL)
4. Database time (provided by ActiveRecord instrumentation and others)
5. Rendering time (nothing provided, would be provided by ActionView instrumentation)

Adding this effectively helps users see number 5.

File: lib/ddtrace/contrib/action_view/events/render_template.rb (hunk 4 above)
Comment (delner) [reply lib/ddtrace/contrib/action_view/events/render_template.rb]: Same as other answer.

File: lib/ddtrace/contrib/action_view/events/render_partial.rb (hunk 3 above)
Comment (brettlangdon) [reply lib/ddtrace/contrib/action_view/events/render_partial.rb]: k

Comment (brettlangdon) [reply lib/ddtrace/contrib/action_view/events/render_partial.rb]: makes sense, as long as we are adding it purposefully

**Pull request merged**